src/puppetlabs/puppetdb/http/server.clj

```diff
@@ -29,30 +27,23 @@
     (log/warn msg)
     (header result "Warning" msg)))
 
-(defn deprecated-v2-app
-  [request]
-  (deprecated-app
-   v2-app
-   "v2 query API is deprecated and will be removed in an upcoming release.  Please upgrade to v3."
-   request))
+(defn- refuse-retired-api
+  [version]
+  (constantly
+   (http/error-response
+    (format "The %s API has been retired; please use v4" version)
```

Comment:
I'm fine either way -- I was just matching the behavior of another 404 example in the code.